src/hypertable.c

```diff
@@ -2551,6 +2551,39 @@ ts_hypertable_create_compressed(Oid table_relid, int32 hypertable_id)
 	Relation rel;
 
 	rel = table_open(table_relid, AccessExclusiveLock);
+	int32 row_size = MAXALIGN(SizeofHeapTupleHeader);
+	/* estimate tuple width of compressed hypertable */
+	for (int i = 1; i <= RelationGetNumberOfAttributes(rel); i++)
+	{
+		bool is_varlena = false;
+		Oid outfunc;
+		Form_pg_attribute att = TupleDescAttr(rel->rd_att, i - 1);
+		if (att->attisdropped)
+			continue;
+		getTypeOutputInfo(att->atttypid, &outfunc, &is_varlena);
+		if (is_varlena)
+			row_size += 18;
+		else
+			row_size += att->attlen;
+	}
+	if (row_size > MaxHeapTupleSize)
+	{
+#ifndef NDEBUG
+		ereport(WARNING,
+				(errmsg("compressed row size might exceed maximum row size"),
+				 errdetail("Estimated row size of compressed hypertable is %u. This exceeds the "
+						   "maximum size of %u and can cause compression of chunks to fail.",
+						   9956,
+						   8164)));
+#else
```

Comment:
I am not sure why you need a separate case for when `NDEBUG` is not defined.

Reply:
On some platforms the maximum row size returned is 8160 and on some its 8164. This causes tests to fail.
Thus hardcoding values on debug specific builds.

Reply:
But `NDEBUG` is the symbol used to enable or disable `assert`. Also, since you have put the numbers in the detailed message, you can set the verbosity to "terse" to exclude printing the detailed error message. See [Error Reporting and Logging](https://www.postgresql.org/docs/current/runtime-config-logging.html) and [psql](https://www.postgresql.org/docs/14/app-psql.html) in the PostgreSQL documentation.

Reply:
"terse" is set as part of log_error_verbosity which controls what should be logged in server logs.
In the tests, messages are sent to client (psql).
client_min_messages default is (NOTICE) even if i set to WARNING, detailed message is sent by server.

Code block defined inside NDEBUG executes only in debug builds i suppose.
Current patch i removed NDEBUG block and i see this error diffs.


@@ -628,7 +628,7 @@
 ALTER TABLE ts_table SET(timescaledb.compress, timescaledb.compress_segmentby = 'c1',
 					   timescaledb.compress_orderby = 'tm');
 WARNING:  compressed row size might exceed maximum row size
-DETAIL:  Estimated row size of compressed hypertable is 9952. This exceeds the maximum size of 8164 and can cause compression of chunks to fail.
+DETAIL:  Estimated row size of compressed hypertable is 9952. This exceeds the maximum size of 8160 and can cause compression of chunks to fail.

Reply:
You can use `\set VERBOSITY terse` to skip the error details in the test. Check for this in the other test files.

Reply:
Fixed.